src/python/parser.py

```diff
@@ -130,6 +143,13 @@ def parse_block(self):
         while not self.consume(T.RIGHT_CURLY) and self.cur < len(self.tokens):
             statements.append(self.parse_statement())
         return Node(BLOCK, statements)
+    
+    def parse_struct_block(self):
+        fields = []
+        self.consume(T.LEFT_CURLY)
+        while not self.consume(T.RIGHT_CURLY) and self.cur < len(self.tokens):
+            fields.append(self.parse_field_declr())
+        return Node(STRUCT_BLOCK, fields)
```

Comment:
Probably better to have this inlined in `parse_struct`. I know they are separate in the grammar (and that can be kept), but it would simplify the tree a bit.